omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -545,30 +545,24 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                 }
             });
 
-            // Browser back button support, for preserve data
-            if (lastSearchParams !== null) {
-                $j('#inputNode').val(lastSearchParams.lastSearchText);
-                if (lastSearchParams.includeVoided == 1) {
-                    $j('#includeVoided').attr('checked','checked');
-                }
-                if (lastSearchParams.includeVerbose == 1) {
-                    $j('#includeVerbose').attr('checked','checked');
-                }
-                var keyEvent = jQuery.Event("keyup");
-
-                var codex = 13;
-
-                // Patient search doesn't support for Enter Key. So invoke keyup
-                // event with last character of searched text.
-                if (el.attr('id') === "findPatients") {
-                    var text = $j('#inputNode').val();
-                    codex = $j('#inputNode').val().charCodeAt($j('#inputNode').val().length - 1);
+            //Browser back button support, to preserve data
+            if(typeof(Storage) !== "undefined") {
+            	//check to see if there are parameters to get from sessionStorage
+            	//after setting the fields clear out session storage so it doesn't get used on other pages
+            	if(sessionStorage.lastSearchText){
+            		self.options.searchPhrase = sessionStorage.lastSearchText;  
+            		sessionStorage.lastSearchText = "";
```

Comment:
You need to remove it instead with the removeItem method,below is how you do it:

`sessionStorage.removeItem("lastSearchText");`